andrewinsoul/RoleManagementSystem
# implement signup of user

#### What does this PR do?
This PR implements the user signup

#### Description of Task to be completed?
- Add util for generating token of user that signed up
- Add util for sending mail to user who successfully signed up
- Fix minor bugs on the model files

#### How should this be manually tested?
- Fetch the branch locally by running `git fetch origin ft-signup-user`
- cd into the server directory by running `cd ./server`
- Run `npm run migrate to create the database tables`
- Using postman, create a payload of type json with in the following format:
{
  firstName: <name>,
  lastName: <name>,
  email: <emai>l,
  password: <password>,
  phone: <phone>,
  roleId: <roleId>
}
A message to check your email will be returned and on checking the inbox of the mail address used in signing up, you should see further instructions on how to verify your account
